libshiboken/basewrapper.cpp

```diff
@@ -770,7 +770,13 @@ bool isUserType(PyObject* pyObj)
 Py_hash_t hash(PyObject* pyObj)
 {
     assert(Shiboken::Object::checkType(pyObj));
+#if __SIZEOF_POINTER__ == 8
```

Comment:
that's gcc only @ViacheslavYartsev looks like it's better to have something like 
```c++
unsigned long result = reinterpret_cast<uintptr_t>(this);
for(int i=sizeof(unsigned long); i<sizeof(uintptr_t); i+=sizeof(unsigned long)) {
  result = result ^  static_cast<unsigned long>(reinterpret_cast<uintptr_t>(this) >> i*8);
}
return result;
```
Since cycle is depend only on compile-time known variables it will be optimized out by compiller